app/src/main/java/org/oppia/android/app/home/HomeActivityPresenter.kt

```diff
@@ -26,6 +30,17 @@ class HomeActivityPresenter @Inject constructor(private val activity: AppCompatA
         TAG_HOME_FRAGMENT
       ).commitNow()
     }
+
+    if (getSpotlightFragment() == null) {
+      val spotlightFragment = SpotlightFragment()
+      val args = Bundle()
+      args.putInt(PROFILE_ID_ARGUMENT_KEY, internalProfileId)
+      spotlightFragment.arguments = args
```

Comment:
Prefer having a create() method within SpotlightFragment that takes the profile ID, and uses its own internal key constant for adding/retrieval. This is what we do elsewhere.

Ditto everywhere in the PR.

Reply:
could you please provide a reference i could use? For all the create functions i saw in the app are for viewmodel's factory class methods.

Reply:
@JishnuGoyal For clarification @BenHenning is talking about `newInstance` https://github.com/oppia/oppia-android/blob/6f26b9e90b5c2c6e269aa97d8f16b24eb0b3511c/app/src/main/java/org/oppia/android/app/topic/info/TopicInfoFragment.kt#L19

You should create a companion object inside SpotlightFragment named `newInstance` which can be used whenever this fragment needs to be created anywhere.

Reply:
thank you so much Rajat and Ben. This is done, please takea look.